commands/template_store_list.go

```diff
@@ -0,0 +1,170 @@
+)
+
+var (
+	source string
+)
+
+func init() {
+	templateStoreListCmd.Flags().StringVarP(&source, "source", "s", allTemplates, "The source of the templates")
+	templateStoreListCmd.Flags().BoolVarP(&verbose, "verbose", "v", false, "Shows additional language and platform")
+
+	templateStoreCmd.AddCommand(templateStoreListCmd)
+}
+
+var templateStoreListCmd = &cobra.Command{
+	Use:     `list`,
+	Short:   `List templates from OpenFaaS organizations`,
+	Aliases: []string{"ls"},
+	Long:    `List templates supported by openfaas or openfaas-incubator organizations or both`,
+	Example: `  faas-cli template store list --source=openfaas --verbose
+  faas-cli template store ls --source=openfaas-incubator -v
+  faas-cli template store ls --source=all`,
+	RunE: runTemplateStoreList,
+}
+
+func runTemplateStoreList(cmd *cobra.Command, args []string) error {
+	templatesInfo, templatesErr := getTemplateInfo(DefaultTemplateList)
+	if templatesErr != nil {
+		return fmt.Errorf("error while getting templates info: %s\n", templatesErr)
+	}
+
+	formattedOutput := formatTemplatesOutput(templatesInfo)
+
+	fmt.Println(formattedOutput)
+	return nil
+}
+
+func getTemplateInfo(repository string) (TemplatesInfo, error) {
+	req, reqErr := http.NewRequest(http.MethodGet, repository, nil)
+	if reqErr != nil {
+		return nil, fmt.Errorf("error while trying to create request to take template info: %s\n", reqErr.Error())
+	}
+
+	client := http.DefaultClient
+	res, clientErr := client.Do(req)
+	if clientErr != nil {
+		return nil, fmt.Errorf("error while requesting template list: %s\n", clientErr.Error())
+	}
+
+	if res != nil {
+		defer res.Body.Close()
+		if res.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("unexpected status code wanted: %d got: %d\n", http.StatusOK, res.StatusCode)
+		}
+
+		body, bodyErr := ioutil.ReadAll(res.Body)
+		if bodyErr != nil {
+			return nil, fmt.Errorf("error while reading data from templates body: %s\n", bodyErr.Error())
+		}
+
+		templatesInfo := TemplatesInfo{}
+		unmarshallErr := json.Unmarshal(body, &templatesInfo)
+		if unmarshallErr != nil {
+			return nil, fmt.Errorf("error while unmarshalling into templates struct: %s\n", unmarshallErr.Error())
+		}
+		return templatesInfo, nil
+	}
+	return nil, fmt.Errorf("error empty response body from: %s\n", DefaultTemplateList)
+}
+
+func formatTemplatesOutput(templates TemplatesInfo) string {
+	if source != allTemplates && source != incubatorRepo && source != officialRepo {
+		fmt.Printf("\nThe --source flag needs to have value %s, %s or %s\n", allTemplates, incubatorRepo, officialRepo)
+		return ""
+	}
+	var buff bytes.Buffer
+	lineWriter := tabwriter.NewWriter(&buff, 0, 0, 1, ' ', 0)
+	fmt.Fprintln(lineWriter)
+	if verbose {
+		fmt.Fprintf(lineWriter, "NAME\tLANGUAGE\tPLATFORM\tSOURCE\tDESCRIPTION\n")
+	} else {
+		fmt.Fprintf(lineWriter, "NAME\tSOURCE\tDESCRIPTION\n")
+	}
+	for _, template := range templates {
+		if source != allTemplates {
+			if source == incubatorRepo {
```

Comment:
Not sure why we need these three code blocks? I'd expect it to be either a named language, a named repo or `all`

Reply:
This is because the sources are `openfaas` and `openfaas-incubator` by default we set all, but it is a little bit spam, since I expect people to be used in the uncubator so this is a way to filter from where are the templates displayed. Like I want `openfaas-incubator` templates only so i will `--source=openfaas-incubator`